internal/common/rules/RuleMatrix.go

```diff
@@ -0,0 +1,48 @@
+package rules
+
+import "gonum.org/v1/gonum/mat"
+
+type RuleMatrix struct {
+	ruleName          string
+	RequiredVariables []string
+	ApplicableMatrix  mat.Dense
+	AuxiliaryVector   mat.VecDense
+}
+
+var AvailableRules = map[string]RuleMatrix{}
+
+func registerNewRule(ruleName string, requiredVariables []string, applicableMatrix mat.Dense, auxiliaryVector mat.VecDense) *RuleMatrix {
```

Comment:
why return the matrix? is this used?

Reply:
It depends on how our Global/Local rule caching architecture works. I returned it for completeness but can take it out if we are going for global cache.